bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -585,6 +585,10 @@ public static TypedescValue getTypedesc(Object value) {
      * @return the annotation value if present, nil else
      */
     public static Object getAnnotValue(TypedescValue typedescValue, BString annotTag) {
+        MapValue annotMap = typedescValue.getAnnotations();
+        if (annotMap != null) {
```

Comment:
Can we check by type tag instead and specifically do this only for records and tuples?

Reply:
Removed this logic
https://github.com/ballerina-platform/ballerina-lang/pull/38488#discussion_r1083963383